StealFocus/steal-focus-explainer.md

```diff
@@ -0,0 +1,79 @@
+# Prevent iframe to take programmatic focus from itself
+
+Authors: [Siye Liu](https://github.com/siliu1)
+
+## Status of this Document
+This document is a starting point for engaging the community and standards bodies in developing collaborative solutions fit for standardization. As the solutions to problems described in this document progress along the standards-track, we will retain this document as an archive and use this section to keep the community up-to-date with the most current standards venue and content location of future work and discussions.
+* This document status: **`ACTIVE`**
+* Expected venue: [Web Hypertext Application Technology Working Group (WHATWG)](https://whatwg.org/)
+* Current version: **https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/<TO BE FILLED>**
+
+## Introduction
+Web applications can host embedded apps via iframes in the web. The embedded apps may take keyboard focus through script. At the same time, the host application may not want to lose keyboard focus when rendering the content in one embedded iframe as the user could have a negative experience when navigating through multiple embedded apps via keyboard. This proposal aims to provided web applications with the ability to control embedded iframes in such a way that guarantees their users to have a good experience when rendering iframe's content (without giving focus to the embedded iframe).
+
+## Goals
+Propose a mechanism to prevent iframe from being able to take programmtic focus unless:
+- When its parent window set the focus on iframe element explicitly. Or,
+- A user moves focus to the iframe by clicking inside the iframe.
+
+## Non-Goals
+It is not a goal of this proposal to allow render iframes to arbitrarily control focus behavior of the browser.
```

Comment:
I'm not sure if I understand this non-goal. Could you explain why this is an interesting non-goal?